server/mqtt.go

```diff
@@ -3914,7 +4423,6 @@ func (sess *mqttSession) processJSConsumer(c *client, subject, sid string,
 			c.Errorf("Unable to add JetStream consumer for subscription on %q: err=%v", subject, err)
 			return nil, nil, err
 		}
-		sess.tmaxack += maxAckPending
```

Comment:
Hmm... not sure why it is taken out, review

Reply:
Restored the logic and got rid of the `sess.createConsumer` wrapper; use `sess.jsa.createConsumer` since sess is always locked when we do it.